overlord/snapstate/handlers.go

```diff
@@ -1058,6 +1083,12 @@ func snapdSnapInstalled(st *state.State) bool {
 	return err == nil
 }
 
+func coreSnapInstalled(st *state.State) bool {
+	var snapst SnapState
+	err := Get(st, "core", &snapst)
+	return err == nil
```

Comment:
Observation, the original implementation of IsInstalled returns `snapState.IsInstalled(), nil` here, and IsInstalled makes an extra check around current and snap sequence list, we're loosing it here by only checking err - this is probably fine?

Reply:
missing the IsInstalled is fine, but I'm slightly worried that we should return the error from the caller if it's not ErrNoState

Reply:
I can add code that returns the error. It will make the call side of the code a bit unwieldy but it can be added.